### PR TITLE
Halp

### PR DESCRIPTION
**Pull request by gmrabian**
## Description
Brax must eat

## Code author checklist
- [ ] I have performed a self-review of my code
- [ ] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [ ] I have added analytics, if necessary
- [ ] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

**Review by scwambach — APPROVED** [review]

File: services/ui-src/src/utils/validation/schemas.test.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ describe("Schemas", () => {
     "N/A",
     "Data not available",
   ];
-  const badNumberTestCases = ["abc", "N", "", "123:123", "!@#!@%"];
+  const badNumberTestCases = ["abc", "N", "", "!@#!@%"];
```
Comment (braxex) [inline services/ui-src/src/utils/validation/schemas.test.ts:17]: Question: Do we need the `""` test case here or can we remove it altogether?

File: services/ui-src/src/utils/validation/schemas.ts [hunk 2]
```diff
@@ -17,30 +17,28 @@ export const textOptional = () => text().notRequired();
...
     .test({
       message: error.INVALID_NUMBER_OR_NA,
-      test: (val) => {
-        const replaceCharsRegex = /[,.]/g;
-        return (
-          ignoreCharsForSchema(val, replaceCharsRegex).isValidSync(val) ||
-          validNAValues.includes(val)
-        );
+      test: (value) => {
+        const validNumberRegex = /[0-9,.]/;
```
Comment (braxex) [inline services/ui-src/src/utils/validation/schemas.ts:32]: Info: We could replace `0-9` with `d` to be real regexy, but honestly I like 0-9 better. Why do I bring it up? Idk.

File: services/ui-src/src/utils/validation/schemas.ts [hunk 3]
```diff
@@ -17,30 +17,28 @@ export const textOptional = () => text().notRequired();
...
+        if (value) {
+          const isValidStringValue = !!validNAValues.includes(value);
+          const isValidNumberValue = !!value?.match(validNumberRegex);
```
Comment (braxex) [inline services/ui-src/src/utils/validation/schemas.ts:35]: Suggestion: We can get just a boolean from a regex test by using `regex.test` instead, like this:

```suggestion
          const isValidStringValue = validNAValues.includes(value);
          const isValidNumberValue = validNumberRegex.test(value);
```

It has the added bonus of not having to use boolean coalescing.

File: services/ui-src/src/utils/validation/schemas.ts [hunk 4]
```diff
@@ -17,30 +17,28 @@ export const textOptional = () => text().notRequired();
...
+          return isValidStringValue || isValidNumberValue;
+        }
+        return true;
```
Comment (braxex) [inline services/ui-src/src/utils/validation/schemas.ts:38]: Nit: This should really be `else return true`.
```suggestion
        } else return true;
```

**Review by braxex — DISMISSED** [review]
We still need to copy this to the server-side validation schema. Thanks @karla-vm and @gmrabian and @ntsummers1 for all your work on this one.

File: services/ui-src/src/utils/validation/schemas.test.ts (hunk 1 above)
Comment (gmrabian) [reply services/ui-src/src/utils/validation/schemas.test.ts:17]: still a bad number 🤷 we don't want to allow empty string

File: services/ui-src/src/utils/validation/schemas.ts (hunk 2 above)
Comment (gmrabian) [reply services/ui-src/src/utils/validation/schemas.ts:32]: fair point. I like the readability of 0-9

**Review by braxex — APPROVED** [review]
Looking good. 

Does that masking issue happen in main? If no, I think I know the problem. Here's a hint for @gmrabian. Maybe your co-conspirator can help you puzzle it out.

>Monikers given, expressed or guessed,
Is character common? Regular, yes,
In thine own self begin and start at the first,
The answer you'll find: narrow is worst.

Let's make that change and see how the mask works afterward.

**Review by karla-vm — APPROVED** [review]
💯

**Review by karla-vm — APPROVED** [review]

**Pull request merged**